### PR TITLE
remove doomednum reset for MT_SCEPTRE and MT_BIBLE

### PR DESCRIPTION
**Pull request by rfomin**
This change fixes desync in EVIT04-542.lmp, EVIT05-231.lmp, and many other Eviternity demos. There are still a few desyncs left.

### CODE REVIEW

**Comment by fabiangreffrath** [comment]
So, this is the change in MBF's logic that we need to change to match PrBoom+'s logic? Does this mean we will desync with MBF demos?

**Comment by rfomin** [comment]
> So, this is the change in MBF's logic that we need to change to match PrBoom+'s logic?

Yes.

> Does this mean we will desync with MBF demos?

Do you mean DOS MBF?

**Comment by fabiangreffrath** [comment]
Yes, I mean "the real" MBF.exe.

**Comment by rfomin** [comment]
Yes, we will desync with DOS MBF.  As you know Eviternity not run on it properly anyway.  Outside of Eviternity and Valiant, there are very few demos to test.

**Comment by fabiangreffrath** [comment]
Forgot to say: Thank you very much for pointing out this discrepancy, and I am really glad you finally found it!

However, now this leads us into a dilemma. Resetting their doomednum to `-1` means that the map things aren't even spawned, right? Phew...! We will need a compat flag for this, probably even defaulting to not resetting them. Being MBF.exe compatible is still the mission goal, but preventing to spawn map things that are actually used in-game must not stand in its way...

**Comment by rfomin** [comment]
> defaulting to not resetting them

Agree.
Also, I think we should fix P_FindShortestTextureAround and P_FindShortestUpperAround (#126) without comp checking, this is an obvious typo, and currently at least 1024cla2.wad map11 does not work in Woof.

**Comment by fabiangreffrath** [comment]
Let me sleep about it.. But yes, most probably we should do that. PrBoom+ is the reference and it has these fixes since forever. It is probably more important to be compatible with it than with MBF.exe for maps that the DOS version couldn't properly play anyway...

**Comment by fabiangreffrath** [comment]
> There are still a few desyncs left

Any ideas what this may be?

**Comment by rfomin** [comment]
There is definitely something wrong with Eviternity map14. All demos with monsters desync here: EVIT14-839.lmp, EVIT14n359.lmp, evit_m14_uvs_317.lmp ... 
evit_m14_uvs_344.lmp desync probably around 1209 gametic. My current theory is that desync has something to do with voodoo dolls. As you can imagine, I'm a little tired of desync hunting, maybe I'll try later this week. Any help and advice are greatly appreciated.

**Comment by fabiangreffrath** [comment]
You have done a great job hunting bugs and demo desyncs!

Regarding this specific one, I guess it's safe to hide the code in question behind a `-strict` command line flag or similar. In original MBF you had no change to play the map as intended, because some map things simply weren't spawned if you were not in "beta emulation" mode which had caveats on its own. If we want to enable use of extended map thing and states ranges (aka DEHEXTRA) it doesn't make sense to have this restriction enabled by default.

**Comment by rfomin** [comment]
I suggest simply deleting this code. `-strict` is impractical, there are no known maps or demos that depend on it.

File: Source/d_main.c [hunk 1]
```diff
@@ -1497,8 +1497,8 @@ void D_DoomMain(void)
       mobjinfo[MT_SKULL].deathstate   = S_BSKUL_DIE1;
       mobjinfo[MT_SKULL].damage       = 1;
     }
-  else
-    mobjinfo[MT_SCEPTRE].doomednum = mobjinfo[MT_BIBLE].doomednum = -1;
+  // else
```
Comment (fabiangreffrath) [inline Source/d_main.c:1500]: Could you please comment this out with an `if 0 ... endif` range and add a comment?

**Review by fabiangreffrath — CHANGES_REQUESTED** [review]

**Comment by fabiangreffrath** [comment]
Perfect, thanks!

**Pull request merged**